ouroboros-consensus/src/Ouroboros/Consensus/Node/ProtocolInfo/Byron.hs

```diff
@@ -40,46 +48,113 @@ import qualified Test.Cardano.Chain.Genesis.Dummy as Dummy
   Credentials
 -------------------------------------------------------------------------------}
 
-data PbftLeaderCredentials = PbftLeaderCredentials {
-      plcSignKey     :: Cardano.SigningKey
-    , plcDlgCert     :: Cardano.Delegation.Certificate
+data PBftLeaderCredentials = PBftLeaderCredentials {
+      plcSignKey     :: Crypto.SigningKey
+    , plcDlgCert     :: Delegation.Certificate
+    , plcCodeNodeId  :: CoreNodeId
+    } deriving (Eq, Show)
+
+-- | Make the 'PBftLeaderCredentials', with a couple sanity checks:
+--
+-- * That the block signing key and the delegation certificate match.
+-- * That the delegation certificate does correspond to one of the genesis
+--   keys from the genesis file.
+--
+mkPBftLeaderCredentials :: Genesis.Config
+                        -> Crypto.SigningKey
+                        -> Delegation.Certificate
+                        -> Either PBftLeaderCredentialsError
+                                  PBftLeaderCredentials
+mkPBftLeaderCredentials gc sk cert = do
+    guard (Delegation.delegateVK cert == Crypto.toVerification sk)
+      ?! NodeSigningKeyDoesNotMatchDelegationCertificate
+
+    let vkGenesis = Delegation.issuerVK cert
+    nid <- genesisKeyCoreNodeId gc (VerKeyCardanoDSIGN vkGenesis)
+             ?! DelegationCertificateNotFromGenesisKey
+
+    return PBftLeaderCredentials {
+      plcSignKey     = sk
+    , plcDlgCert     = cert
+    , plcCodeNodeId  = nid
     }
+  where
+    (?!) :: Maybe a -> e -> Either e a
+    Just x  ?! _ = Right x
+    Nothing ?! e = Left  e
```

Comment:
@dcoutts, should we add this to the `cardano-prelude` (or otherwise similar place)?